testsuite/testcases/src/quorum_store_onchain_enable_test.rs

```diff
@@ -0,0 +1,131 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::NetworkLoadTest;
+use anyhow::Ok;
+use aptos::test::CliTestFramework;
+use aptos_forge::{NetworkTest, NodeExt, SwarmExt, Test};
+use aptos_logger::info;
+use aptos_sdk::bcs;
+use aptos_types::{
+    account_config::CORE_CODE_ADDRESS,
+    on_chain_config::{ConsensusConfigV1, OnChainConsensusConfig},
+};
+use std::{fmt::Write, time::Duration};
+use tokio::runtime::Runtime;
+
+const MAX_NODE_LAG_SECS: u64 = 360;
+
+pub struct QuorumStoreOnChainEnableTest {}
+
+impl Test for QuorumStoreOnChainEnableTest {
+    fn name(&self) -> &'static str {
+        "quorum-store reconfig enable test"
+    }
+}
+
+impl NetworkLoadTest for QuorumStoreOnChainEnableTest {
+    fn test(
+        &self,
+        swarm: &mut dyn aptos_forge::Swarm,
+        duration: std::time::Duration,
+    ) -> anyhow::Result<()> {
+        let runtime = Runtime::new().unwrap();
+
+        let faucet_endpoint: reqwest::Url = "http://localhost:8081".parse().unwrap();
+        let rest_client = swarm.validators().next().unwrap().rest_client();
+
+        let mut cli = runtime.block_on(async {
+            CliTestFramework::new(
+                swarm.validators().next().unwrap().rest_api_endpoint(),
+                faucet_endpoint,
+                /*num_cli_accounts=*/ 0,
+            )
+            .await
+        });
+
+        std::thread::sleep(duration / 2);
```

Comment:
This will freeze the entire thread, so if this test isn't run on a separate thread, it will just freeze the entire function.  Are you sure this is what you want and not an async sleep?

Just wondering, I know I've messed it up myself so many times.

Reply:
Yeah, it seems we can block in this thread. copy/paste from another test.